tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,215 @@
+"""
+Test to verify concurrent creation and deletion of multiple PVCs
+"""
+import subprocess
+import logging
+import pytest
+
+from ocs import defaults, constants, ocp, exceptions
+from utility.utils import run_cmd
+from ocsci.testlib import tier1, ManageTest
+from resources.pod import get_ceph_tools_pod
+from resources.pvc import PVC
+from tests.fixtures import (
+    create_rbd_storageclass, create_ceph_block_pool, create_rbd_secret
+)
+
+log = logging.getLogger(__name__)
+PVC_OBJS = []
+TEST_PROJECT = 'test-project'
+PROJECT = ocp.OCP(kind='Project', namespace=TEST_PROJECT)
+
+
+@pytest.fixture()
+def test_fixture(request):
+    """
+    Setup and teardown
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Create new project
+    Create PVCs
+    """
+    # Create new project
+    assert PROJECT.new_project(TEST_PROJECT), (
+        f'Failed to create new project {TEST_PROJECT}'
+    )
+
+    # Create 100 PVCs
+    create_multiple_pvc(
+        self.pvc_base_name, self.number_of_pvc, self.sc_obj.name
+    )
+    log.info(f'Created initial {self.number_of_pvc} PVCs')
+    self.pvc_objs_initial = PVC_OBJS[:]
+    PVC_OBJS.clear()
+
+    # Verify PVCs are Bound and fetch PV names
+    for pvc in self.pvc_objs_initial:
+        pvc.reload()
+        assert pvc.status == constants.STATUS_BOUND, (
+            f'PVC {pvc.name} is not Bound'
+        )
+        self.initial_pvs.append(pvc.backed_pv)
+    log.info(f'Initial {self.number_of_pvc} PVCs are in Bound state')
+
+
+def teardown(self):
+    """
+    Delete PVCs
+    Delete project
+    """
+    # Delete newly created PVCs
+    command = (
+        f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
+        f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done'
+    )
+    proc = run_async(command)
+    assert proc, (
+        f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
+    )
+
+    # Verify command to delete PVCs
+    ret, _, _ = proc.async_communicate()
+    assert not ret, "Deletion of newly created PVCs failed"
+    log.info(f'Newly created {self.number_of_pvc} PVCs are now deleted.')
+
+    # Switch to default project
+    run_cmd(f'oc project {defaults.ROOK_CLUSTER_NAMESPACE}')
+
+    # Delete project created for the testcase
+    PROJECT.delete(resource_name=TEST_PROJECT)
+
+
+def create_multiple_pvc(pvc_base_name, number_of_pvc, sc_name):
```

Comment:
move to pvc.py as a helper function.
if you can also add an option to create in async way it will be great

Reply:
Moved to pvc.py PR #236
Creating in async way can be done by using run_async in utils.py by passing a command to create more than one pvc. Here the implementation is using PVC class. So creating in async way is not implemented in PR 236.

Reply:
Done. PR 236 is merged.